### DISCUSSION TITLE
Filament select is too slow when having a large number of data

### DISCUSSION THREAD

**Discussion by AHMED-GAMAL-AG** [filamentphp/filament#13390] [Help]
### Package

Panel builder

### Package Version

3.2

### How can we help you?

When getting a large amount of data (60k in my case for test and performance purposes) in the Select component, it takes minutes to load, even if I am loading the data from the cache using Redis driver and not directly from the database. Any ideas on how to enhance the performance?

![image](https://github.com/filamentphp/filament/assets/76778937/c63a1452-acad-452b-a462-7d356bd00e60)

the code 

```
                Select::make('user_id')
                    ->live()
                    ->required()
                    ->searchable()
                    ->optionsLimit(20)
                    ->label(__('Pioneer name'))
                    ->hidden(fn (Get $get) => !$get('health_unit_id'))
                    ->options(static fn (Get $get) => User::getPioneerOptions($get('health_unit_id')))
                    ->required(),
```

```
    public static function getPioneerOptions(?string $health_unit = null): Collection
    {
        if ($health_unit === null) {
            return collect();
        }

        if (!Cache::has('pioneers')) {
            return Cache::remember('pioneers', 60, function () use ($health_unit) {
                return self::where('health_unit_id', $health_unit)->get()->pluck('name', 'id');
            });
        } else {
            return Cache::get('pioneers');
        }
    }
```

**Comment by NazarAli** [comment]
لدي نفس المشكلة

**Comment by vikijel** [comment]
It feels slow even with 5 items in the db table (using a relationship based options tho).

I have noticed it is showing the "Loading..." onchange of the search text, but the search request is debounced 500ms by default.

So there is 500ms delay between it showing "Loading..." message and actually sending a request (can see this when i have wire:loading indicator on the page).

Can use ->searchDebounce(250) to shorten the delay, but it should show the "loading..." on request, not onchange to make it obvious when it starts loading.

It also makes request to get the label on each search, which doesn't seem right (doc said it does that once for preselected value only)

**Comment by tominal** [comment]
> When getting a large amount of data (60k in my case for test and performance purposes) in the Select component, it takes minutes to load, even if I am loading the data from the cache using Redis driver and not directly from the database. Any ideas on how to enhance the performance?

Have you tried using a search engine to speed things up like Elasticsearch or Meilisearch? Alternatively, are you sure the query is checking Redis and not performing that SQL query every time? Perhaps the local environment is skipping the cache?

**Reply by tobya** [reply]
This shouldn't matter.  The query even on 60K records should be very fast.  Something else has gone wrong.
